### PR TITLE
[WIP] Osx support

### PR DESCRIPTION
**Pull request by dmeltzer**
Hello,

This is rough and incomplete because my makefile foo is pretty not great and I didn't want to try and figure out conditionals. However, it does have the changes necessary to make midimonster at least partially work on os x. 

Changes:

- Remove midi+evdev (should be conditional, at least for midi we should look for an osx replacement solution..)
- clang requires linker flag -undefined dynamic_lookup so it doesn't complain about not seeing symbols when building libraries.
- For whatever reason with clang, the midimonster target needs to be passed the c file instead of the header.  passing the header also passes the c file, and then we get errors due to multiple targets on command line with -o.
- Define htobe/etc as mac equivilents.  Found here: https://github.com/zeromq/zmqpp/issues/164

One outstanding issue is that binding needs a specific address in config files.  bind * 8000 does not work: Failed to get socket info for * port 8000: nodename nor servname provided, or not known

I may keep poking at this over the weekend, but if not wanted to share my efforts so far.

Thanks for your work on the underlying program, it's fantastic!

### CODE REVIEW

**Comment by cbdevnet** [comment]
Hi, thanks for the effort!

A quick google brought up that there's a way to install libevdev on OSX (though I don't know if `uinput` is supported there), so there may be a way to support the `evdev` backend at least for input. For the current status, disabling it works of course.

I'll have to look into how to handle having platform-dependent makefiles (which is going to be required as some backends are entirely platform-specific, such as the `evdev` backend, and some are at least platform-API-dependent like MIDI).

I guess the easiest way to go forward quickly would be to have multiple makefiles, suffixed with their supported OS (so `makefile.macos`, `makefile.windows`, etc) where required.

Once again, thank you for contributing! Keep up the great work :)

**Comment by cbdevnet** [comment]
I've just changed the evdev backend so that it can be built without output support.
The `uinput` kernel module and headers should no longer be required when adding `-DEVDEV_NO_UINPUT` to the backend modules `CFLAGS`.

**Comment by cbdevnet** [comment]
Ah well sorry, seems I misread: it actually was `libev` which is something else entirely - my bad :dagger: 
Still, the flag might be useful to someone ;)

File: makefile [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 PLUGINDIR = "\"./\""
 
 CFLAGS ?= -g -Wall
 #CFLAGS += -DDEBUG
 %.so: CFLAGS += -fPIC
-%.so: LDFLAGS += -shared
+%.so: LDFLAGS += -shared -undefined dynamic_lookup
 
 midimonster: LDLIBS = -ldl
 midimonster: CFLAGS += -rdynamic -DPLUGINS=$(PLUGINDIR)
```
Comment (cbdevnet) [inline makefile]: I think clang might require `-Wl,-export_dynamic` instead of GCC's `-rdynamic`

**Comment by dmeltzer** [comment]
Thoughts about using cmake or similar to generate the makefile? Would add a dependency, but it would remove the multiple manual makefile issues.

Also, not sure where you develop normally, but I find myself needing to rename makefile to Makefile on Linux to get make to work out of the box

**Comment by cbdevnet** [comment]
Mmh, not a big fan of cmake in general... I'll think about it.

I'm on Debian stable with GNUmake, which tests for `makefile`, `Makefile` and `GNUmakefile`, but I'd be OK with using `Makefile` :)

**Comment by cbdevnet** [comment]
At least according to Travis, OSX compilation works :)

**Pull request closed without merging**